ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -40,97 +40,171 @@ const styles = reactStyles({
+      <label htmlFor='runtime-ram' style={{marginRight: '.25rem'}}>RAM (GB)</label>
       <Dropdown id='runtime-ram'
-                options={fp.flow(
-                  // Show valid memory options as constrained by the currently selected CPU.
-                  fp.filter(({cpu: availableCpu}) => availableCpu === cpu),
-                  fp.map('memory'),
-                  // See above comment on CPU union.
-                  fp.union([memory]),
-                  fp.sortBy(fp.identity)
-                )(validLeonardoMachineTypes)}
-                onChange={
-                  ({value}) => fp.flow(
-                    fp.find({cpu, memory: value}),
-                    // If the selected machine is not different from the current machine return null
-                    maybeGetMachine,
-                    onChange
-                    )(validLeonardoMachineTypes) }
-                value={memory}
-                />
-    </div>
+        options={fp.flow(
+          // Show valid memory options as constrained by the currently selected CPU.
+          fp.filter(({cpu: availableCpu}) => availableCpu === cpu),
+          fp.map('memory'),
+          // See above comment on CPU union.
+          fp.union([memory]),
+          fp.sortBy(fp.identity)
+        )(validLeonardoMachineTypes)}
+        onChange={
+          ({value}) => fp.flow(
+            fp.find({cpu, memory: value}),
+            // If the selected machine is not different from the current machine return null
+            maybeGetMachine,
+            onChange
+            )(validLeonardoMachineTypes) }
+        value={memory}
+        />
   </Fragment>;
 };
 
-const DiskSizeSelection = ({onChange, updatedDiskSize, masterDiskSize}) => {
-  return <div>
-    <label htmlFor='runtime-disk'
-          style={{marginRight: '.25rem'}}>Disk (GB)</label>
-      <InputNumber id='runtime-disk'
-                showButtons
-                decrementButtonClassName='p-button-secondary'
-                incrementButtonClassName='p-button-secondary'
-                value={updatedDiskSize || masterDiskSize}
-                inputStyle={{padding: '.75rem .5rem', width: '2rem'}}
-                onChange={({value}) => onChange(value === masterDiskSize ? null : value)}
-                min={50 /* Runtime API has a minimum 50GB requirement. */}/>
-  </div>;
+const DiskSizeSelection = ({onChange, updatedDiskSize, diskSize}) => {
+  return <Fragment>
+    <label htmlFor='runtime-disk' style={{marginRight: '.25rem'}}>Disk (GB)</label>
+    <InputNumber id='runtime-disk'
+      showButtons
+      decrementButtonClassName='p-button-secondary'
+      incrementButtonClassName='p-button-secondary'
+      value={updatedDiskSize || diskSize}
+      inputStyle={{padding: '.75rem .5rem', width: '2rem'}}
+      onChange={({value}) => onChange(value)}
+      min={50 /* Runtime API has a minimum 50GB requirement. */}/>
+  </Fragment>;
+};
+
+const DataProcConfig = ({onChange, dataprocConfig}) => {
```

Comment:
Please document the semantics somewhere. Specifically - onChange(null) as a sentinel to indicate "no change from current runtime" requires deep code inspection to understand.